nixos/modules/services/networking/seafile.nix

```diff
@@ -0,0 +1,272 @@
+        if [ ! -d ${ccnetDir} ]; then
+            mkdir -p ${ccnetDir}/{GroupMgr,misc,OrgMgr,PeerMgr}
+            ${pkgs.sqlite}/bin/sqlite3 ${ccnetDir}/GroupMgr/groupmgr.db ".read ${pkgs.seafile-server}/share/seafile/sql/groupmgr.sql"
+            ${pkgs.sqlite}/bin/sqlite3 ${ccnetDir}/misc/config.db ".read ${pkgs.seafile-server}/share/seafile/sql/config.sql"
+            ${pkgs.sqlite}/bin/sqlite3 ${ccnetDir}/OrgMgr/orgmgr.db ".read ${pkgs.seafile-server}/share/seafile/sql/org.sql"
+            ${pkgs.sqlite}/bin/sqlite3 ${ccnetDir}/PeerMgr/usermgr.db ".read ${pkgs.seafile-server}/share/seafile/sql/user.sql"
+            ${pkgs.sqlite}/bin/sqlite3 ${dataDir}/seafile.db ".read ${pkgs.seafile-server}/share/seafile/sql/seafile.sql"
+        fi
+      '';
+    };
+
+    systemd.services.seahub =
+      let penv = (pkgs.python3.withPackages (ps: with ps; [ gunicorn seahub ]));
+      in {
+        description = "Seafile Server Web Frontend";
+        wantedBy = [ "seafile.target" ];
+        partOf = [ "seafile.target" ];
+        after = [ "network.target" "seaf-server.service" ];
+        requires = [ "seaf-server.service" ];
+        restartIfChanged = true;
+        restartTriggers = [ (seafileConf) ];
+        environment = {
+          PYTHONPATH =
+            "${pkgs.python3Packages.seahub}/thirdpart:${pkgs.python3Packages.seahub}:${penv}/${python.sitePackages}";
+          DJANGO_SETTINGS_MODULE = "seahub.settings";
+          CCNET_CONF_DIR = ccnetDir;
+          SEAFILE_CONF_DIR = dataDir;
+          SEAFILE_CENTRAL_CONF_DIR = centralConfigDir;
+          SEAFILE_RPC_PIPE_PATH = "/run/seafile";
+        };
+        serviceConfig = {
+          User = "seafile";
+          Group = "seafile";
+          DynamicUser = true;
+          ProtectHome = true;
+          PrivateUsers = true;
+          PrivateDevices = true;
+          ProtectClock = true;
+          ProtectHostname = true;
+          ProtectProc = "invisible";
+          ProtectKernelModules = true;
+          ProtectKernelTunables = true;
+          ProtectKernelLogs = true;
+          ProtectControlGroups = true;
+          RestrictNamespaces = true;
+          RestrictAddressFamilies = [ "AF_UNIX" ];
+          LockPersonality = true;
+          RestrictRealtime = true;
+          RestrictSUIDSGID = true;
+          MemoryDenyWriteExecute = true;
+          SystemCallArchitectures = "native";
+          RuntimeDirectory = "seahub";
+          StateDirectory = "seafile";
+          LogsDirectory = "seafile";
+          ExecStart = ''
+            ${penv}/bin/gunicorn seahub.wsgi:application \
+            --name seahub \
+            --workers ${toString cfg.workers} \
+            --log-level=info \
+            --preload \
+            --timeout=1200 \
+            --limit-request-line=8190 \
+            --bind unix:/run/seahub/gunicorn.sock
+          '';
+        };
+        preStart = ''
+          if [ ! -d "${seahubDir}" ]; then
+              mkdir ${seahubDir}
```

Comment:
As above, probably best handled by `StateDirectory`.

Reply:
Yes, this was used as a semaphore to trigger initial configuration of the server but this should be better done with a dedicated file in data directory.

Reply:
Sounds good :+1: